src/python/antares_xpansion/input_checker.py

```diff
@@ -348,82 +495,71 @@ def check_setting_option_value(option, value):
         :return: True if the option has the correct type, exits if the value has the wrong type
     """
 
-    options_legal_values = {
-        "method": ["benders_decomposition"],
-        "uc_type": ["expansion_accurate", "expansion_fast"],
-        "master": ["relaxed", "integer", "full_integer"],
-        "optimality_gap": None,
-        "relative_gap": None,
-        "cut_type": ["average", "yearly", "weekly"],
-        "week_selection": ["true", "false"],
-        "max_iteration": None,
-        "relaxed_optimality_gap": None,
-        "timelimit": None,
-        "yearly-weights": None,
-        "additional-constraints": None,
-        "log_level": None,
-    }
-    legal_values = options_legal_values.get(option)
+    if not check_setting_option_type(option, value):
+        flushed_print(
+            "check_settings : value %s for option %s has the wrong type!" % (value, option))
+        raise OptionTypeError
+
+    legal_values = options_types_and_legal_values.get(option)[1]
 
     skip_verif = ["yearly-weights", "additional-constraints", "solver"]
```

Comment:
Add deprecated options in `skip_verif`, there is only `master` and `uc_type` left to check

Reply:
what are the deprecated option ?